### PR TITLE
gh-119574: Add some missing environment variables to '--help-env'.

### PR DESCRIPTION
**Pull request by devdanzin**
Add some very basic descriptions of environment variables that were missing from `initconfig.c`.

* Issue: gh-119574

### CODE REVIEW

File: Python/initconfig.c [hunk 1]
```diff
@@ -260,6 +261,20 @@ static const char usage_envvars[] =
 "                  various kinds of output.  Setting it to 0 deactivates\n"
 "                  this behavior.\n"
 "PYTHON_HISTORY  : the location of a .python_history file.\n"
+"PYTHONASYNCIODEBUG: enable asyncio debug mode\n"
+#ifdef Py_DEBUG
```
Comment (encukou) [inline Python/initconfig.c]:
```suggestion
#ifdef Py_TRACE_REFS
```

File: Python/initconfig.c [hunk 2]
```diff
@@ -260,6 +261,20 @@ static const char usage_envvars[] =
...
+"PYTHONDUMPREFS  : dump objects and reference counts still alive after shutdown\n"
+"PYTHONDUMPREFSFILE: dump objects and reference counts to the specified file\n"
+#endif
+#ifdef __APPLE__
+"PYTHONEXECUTABLE: set sys.argv[0] to this value (macOS only)\n"
```
Comment (encukou) [inline Python/initconfig.c:270]: It's the other way around: *Python* sets `PYTHONEXECUTABLE` on macOS. IMO, it shouldn't be on this list.

Comment (devdanzin) [reply Python/initconfig.c:270]: Thank you for reviewing this! I'll remove it then.

Do you know why the following test passes on macOS? Is it being skipped?

```
    @unittest.skipUnless(sys.platform == 'darwin', 'PYTHONEXECUTABLE only works on macOS')
    def test_python_executable(self):
        code = 'import sys; print(sys.executable)'
        expected = "/busr/bbin/bpython"
        rc, out, err = assert_python_ok('-c', code, PYTHONEXECUTABLE=expected)
        self.assertIn(expected.encode(), out)
```

Comment (encukou) [reply Python/initconfig.c:270]: Oh! My bad, I misread the code, and didn't start up a Mac to actually test. Sorry!

I'll revert the latest commit and merge.

Comment (devdanzin) [reply Python/initconfig.c:270]: No problem, thank you again for reviewing, editing and merging!

**Pull request merged**